### ISSUE TITLE
The weight class does not match with the expected type (java.lang.Double)

### ISSUE THREAD

**Issue by Jonty800** [gephi/gephi#1951]
## Expected Behavior
In method "checkWeightDynamicType()" of EdgeImpl.java, an exception is thrown only if the edge weight type is equal to Double.class with the error:

> The weight class does not match with the expected type (java.lang.Double)

The weight class **does** match the expected type. Is this a mistake?

## Current Behavior
Unable to use dynamic edges with a double weight with Interval.class

## Possible Solution
Maybe the code should be 
`if (graphStore != null && !Double.class.equals(graphStore.configuration.getEdgeWeightType())) {`
(Note the "!")
## Steps to Reproduce

`Configuration configuration = graphModel.getConfiguration(); 
configuration.setEdgeWeightType(Double.class); 
graphModel.setConfiguration(configuration);`

`edge = graphModel.factory().newEdge(node1, node2, true); 
Interval interval = new Interval(interval,interval+1); 
edge.setWeight(weight, interval);`

## Context

I'm trying to create a dynamic graph using edge weights over time, using the Gephi Toolkit
## Your Environment

* Version used: Gephi 0.9.3 SNAPSHOT
* Java version: 9
* Operating System: Windows 10

> Exception in thread "main" java.lang.IllegalArgumentException: The weight class does not match with the expected type (java.lang.Double)
	at org.gephi.graph.impl.EdgeImpl.checkWeightDynamicType(EdgeImpl.java:428)
	at org.gephi.graph.impl.EdgeImpl.setTimeWeight(EdgeImpl.java:111)
	at org.gephi.graph.impl.EdgeImpl.setWeight(EdgeImpl.java:107)

**Comment by eduramiba** [comment]
Hi,
To set a dynamic weight (calling ```setWeight``` with interval or timestamp), you must set the configuration ```EdgeWeightType``` to a dynamic type (interval or timestamp).

Check how it's done in gephi:

https://github.com/gephi/gephi/blob/5a789b169d5f12423f5d0584db39baea48598160/modules/ProcessorPlugin/src/main/java/org/gephi/io/processor/plugin/DefaultProcessor.java#L124

**Comment by Jonty800** [comment]
Thanks, this appears to have fixed the issue :)

**Comment by hobbitbr** [comment]
Hey... I'm having a problem with dynamic weight... I tried to use, gephi 0.9.1, 0.9.2 and 0.9.3. 
Before I import the csv. documents I change to timestamp and on edge area I convert a column to dynamic.
So I don't have problems to import nodes csv., but when I try to import the edge csv. if i put to timestampdoublemap gephi said my data has problems (I'm using this format <[2006.0, 2007.0, 3.0), [2007.0, 2008.0, 4.0]>). If i change to interval to introduce the edge csv. I have the java,lang.double problems.
I tried to use the interval but I have problems with nodes csv. (the model of data I used <[2015.0]> TIMESTAMP  AND<[2015.0, 2]> SCORE.

I tried to change the datas presentation with not sucess. 
My ideia is the nodes are going to be dynamic during the time, and de edges need to change the intensity of the relations.
